pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,6 +382,58 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
+@register_useless
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@node_rewriter([Subtensor])
+def local_replace_slice(fgraph, node):
```

Comment:
Could we combine this rewrite with the useless slice one as we already have to iterate over all indexes?

I'm thinking we could keep an index of the last non slice(None) idx. Then at the end we would do `x[tuple(new_idxs)[:last_non_slice_idx]`